### PR TITLE
Person student teacher

### PR DESCRIPTION
**Pull request by Flunavgt**
On this PR:

-   Create class Person with the following:

> Instance vars: @id, @name, and @age.
> Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
> Getters for @id, @name, and @age.
> Setters for @name and @age.
> Private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise.
>  Public method can_use_services? that returns true if person is of age or if they have permission from parents.

Create class Student with the following:
> Inherits from Person.
> Constructor extends parent's constructor by adding @classroom and a parameter for it.
> Method play_hooky that returns "¯\(ツ)/¯".

- Create class Teacher with the following:
> Inherits from Person.
> Constructor extends parent's constructor by adding @specialization and a parameter for it.
> Override can_use_services? so it always returns true.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    true if is_of_age(@age) == true || parent_permission == true
+  end
```
Comment (HAMMAS-SALEEM) [inline person.rb]: - [x] This method can_use_services? should be out of your private methods section. Means it should have been created before line 12 👍 to make it accessible for the user 😊
- [x] In line 19, you are using **is_of_age** method which doesn't exist. You have made method which is **of_age?**. Kindly use that 👍

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+class Student < Person
+  attr_reader :classroom, :parent_permission
+
+  def initialize(age, name = 'unknown', parent_permission: true)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+    super
+    @classroom = classroom
```
Comment (HAMMAS-SALEEM) [inline student.rb]: - [x] Kindly use your super to send your parameters to the parent class as shown below 👍 
 
⬇️
`super(@age, @name, @parent_permission)`

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+class Teacher < Person
+  attr_reader :specialization
+
+  def initialize(specialization, age, name = 'unknown', parent_permission: true)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+    super
+    @specialization = specialization
```
Comment (HAMMAS-SALEEM) [inline teacher.rb]: - [x] Kindly use your super to send your parameters to the parent class as shown below 👍 
 
⬇️
`super(@age, @name, @parent_permission)`

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @Flunavgt :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @Flunavgt, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**